mycroft/tts/tts.py

```diff
@@ -370,7 +371,18 @@ def _execute(self, sentence, ident, listen):
                 #  of the TTS cache.  But this requires changing the public
                 #  API of the get_tts method in each engine.
                 audio_file = self.cache.define_audio_file(sentence_hash)
-                _, phonemes = self.get_tts(sentence, str(audio_file.path))
+                # TODO 21.08: remove mutation of audio_file.path.
+                returned_file, phonemes = self.get_tts(
+                    sentence, str(audio_file.path))
+                if returned_file.path != audio_file.path:
```

Comment:
returned file is a string, not a cached object, this check will throw an exception